src/rules/indentRule.ts

```diff
@@ -121,6 +128,161 @@ function walk(ctx: Lint.WalkContext<Options>): void {
+    const tabs: boolean = params.tabs;
+    const size: number = params.size;
+    const pos: number = params.pos;
+    const indentEnd: number = params.indentEnd;
+    const line: string = params.line;
+    const sourceFile: ts.SourceFile = params.sourceFile;
+
+    const currentNodePos: number = pos + indentEnd;
+    let currentNode: ts.Node | undefined = recursiveGetNodeAt(sourceFile, currentNodePos);
+    let depth = 0;
+    if (currentNode !== undefined && inTsx(sourceFile)) {
+        return false;
+    }
+    if (currentNode !== undefined && currentNode.getStart() === currentNodePos) {
+        depth = getNodeDeepth(currentNode, line);
+    } else {
+        currentNode = recursiveGetNodeAtEnd(sourceFile, currentNodePos);
+        if (currentNode !== undefined) {
+            depth = getNodeDeepth(currentNode, line);
+        } else {
+            return false;
+        }
+    }
+    const expectedIndentation: string = tabs ? "\t".repeat(depth)
+            : " ".repeat(size * depth);
+    const actualIndentation = line.slice(0, indentEnd);
+    const failure = Rule.FAILURE_STRING(tabs ? "tab" : size === undefined ? "space" : `${depth * size} space`);
+    if (inStringTemplate(currentNode)) {
+        return false;
+    }
+    if (line.trim().charAt(0) === "*") {
+        return false;
+    }
+    if (expectedIndentation !== actualIndentation) {
+        ctx.addFailureAt(
+            pos,
+            indentEnd,
+            failure,
+            createIndentSizeFix(
+                pos,
+                tabs,
+                size,
+                depth,
+                indentEnd,
+            ),
+        );
+        return true;
+    }
+    return false;
+}
+
+function recursiveGetNodeAt(root: ts.Node, pos: number): ts.Node | undefined {
+    let result: ts.Node | undefined = root.getChildAt(pos);
+    if (result === undefined) {
+        ts.forEachChild(root, function cb(node: ts.Node): void {
+            if (node.getFullStart() < pos && node.getEnd() >= pos) {
+                result = recursiveGetNodeAt(node, pos);
+            }
+            ts.forEachChild(node, cb);
+        });
+    }
+    if (result !== undefined) {
+        return result;
+    }
+    return root;
+}
+
+function recursiveGetNodeAtEnd(root: ts.Node, pos: number): ts.Node | undefined {
+    if (root.getEnd() === pos) {
+        return root;
+    }
+    let result: ts.Node | undefined;
+    ts.forEachChild(root, function cb(node: ts.Node): void {
+        if (node.getFullStart() < pos && node.getEnd() >= pos) {
+            result = recursiveGetNodeAt(node, pos);
+        }
+        ts.forEachChild(node, cb);
+    });
+    return result;
+}
+
+function inTsx(node: ts.SourceFile): boolean {
+    return /.tsx$/.test(node.fileName);
+}
+
+function getNodeDeepth(node: ts.Node, line: string): number {
+    let result = 0;
+    let parent: ts.Node | undefined = node.parent;
+    const blockTypes: ts.SyntaxKind[] = [
+        ts.SyntaxKind.Block, // 207
```

Comment:
these numbers are subject to change with future versions of typescript. you can remove them, because they do not add any value

Reply:
removed